mgr/src/server/datastore/roles.cr

```diff
@@ -12,8 +12,8 @@ module Datastore
   end
 
   def viewable_volume_ids(user_id, pool_id)
-    query = "SELECT DISTINCT volume_id FROM roles WHERE user_id = ? AND pool_id = ?"
-    connection.query_all(query, user_id, as: String)
+    query = "SELECT DISTINCT volume_id FROM roles WHERE user_id = ? AND (pool_id = ? OR pool_id = ?)"
+    connection.query_all(query, user_id, pool_id, "all", as: String)
```

Comment:
Is this because sometimes a user has permission to view all pools?
If user has permission to view `all` pools, will exact pool_id not be present?

Reply:
Roles table will have `user_id`, `pool_id`, `volume_id` and `role`. If a user is Pool admin then that user can see all Volume info and manage. 

- user_id: U1, pool_id: all, volume_id: all, role: admin  - Global Admin across the Pools
- user_id: U1, pool_id: P1, volume_id: all, role: admin - Admin for Pool P1
- user_id: U1, pool_id: P1, volume_id: V1, role: admin - Admin for Volume V1

Possible Roles: Admin, Maintainer, Viewer, Client

Reply:
> If user has permission to view all pools, will exact pool_id not be present?

Yes. Even if it is present then "all" takes the precedence. Ex: A user having permission for both P1 and "all". P1 permission is useless since "all" already covers it.

Reply:
Thanks! Got it!